packages/core/contracts/financial-templates/common/FundingRateApplier.sol

```diff
@@ -0,0 +1,186 @@
+        address _fpFinderAddress,
+        address _timerAddress,
+        bytes32 _identifier
+    ) public Testable(_timerAddress) nonReentrant() {
+        // TODO: Should we constrain `_initialFundingRate > x && < y`?
+        cumulativeFundingRateMultiplier = _initialFundingRate;
+        fpFinder = FinderInterface(_fpFinderAddress);
+        lastUpdateTime = getCurrentTime();
+        identifier = _identifier;
+    }
+
+    /****************************************
+     *         INTERNAL FUNCTIONS           *
+     ****************************************/
+
+    // Returns a token amount scaled by the current funding rate multiplier.
+    // Note: if the contract has paid fees since it was deployed, the raw
+    // value should be larger than the returned value.
+    function _getFundingRateAppliedTokenDebt(FixedPoint.Unsigned memory rawTokenDebt)
+        internal
+        view
+        returns (FixedPoint.Unsigned memory tokenDebt)
+    {
+        return rawTokenDebt.mul(cumulativeFundingRateMultiplier);
+    }
+
+    function _getFundingRateStore() internal view returns (FundingRateStoreInterface) {
+        return FundingRateStoreInterface(fpFinder.getImplementationAddress("FundingRateStore"));
+    }
+
+    // Fetches a funding rate from the Store, determines the period over which to compute an effective fee,
+    // and multiplies the current multiplier by the effective fee.
+    // A funding rate < 1 will reduce the multiplier, and a funding rate of > 1 will increase the multiplier.
+    // Note: 1 is set as the neutral rate because there are no negative numbers in FixedPoint, so we decide to treat
+    // values < 1 as "negative".
+    function _applyEffectiveFundingRatePerToken() internal {
+        uint256 currentTime = getCurrentTime();
+        uint256 paymentPeriod = currentTime.sub(lastUpdateTime);
+
+        FundingRateStoreInterface fundingRateStore = _getFundingRateStore();
+        FixedPoint.Unsigned memory _latestFundingRatePerSecondPerToken = fundingRateStore.getFundingRateForIdentifier(
+            identifier
+        );
+
+        FixedPoint.Unsigned memory effectiveFundingRateForPeriodPerToken;
+        (
+            cumulativeFundingRateMultiplier,
+            effectiveFundingRateForPeriodPerToken
+        ) = _calculateEffectiveFundingRatePerToken(
+            paymentPeriod,
+            _latestFundingRatePerSecondPerToken,
+            cumulativeFundingRateMultiplier
+        );
+        lastUpdateTime = currentTime;
+
+        emit NewFundingRate(
+            cumulativeFundingRateMultiplier.rawValue,
+            lastUpdateTime,
+            paymentPeriod,
+            _latestFundingRatePerSecondPerToken.rawValue,
+            effectiveFundingRateForPeriodPerToken.rawValue
+        );
+    }
+
+    function _calculateEffectiveFundingRatePerToken(
+        uint256 paymentPeriodSeconds,
+        FixedPoint.Unsigned memory fundingRatePerSecondPerToken,
```

Comment:
nit: I think `PerToken` in these names is consistent with how we named them in the store, but do you think this is more effective than `fundingRatePerSecond`? I just sometimes find these long names confusing (I feel the same way when I looks back at the Store code). Just wanted to get y'all's thoughts. Don't have any other alternatives in mind, but definitely open to other, more intuitive naming schemes if y'all have any in mind.

Reply:
I'm good with that, `perToken` is probably implied anyways